packages/eslint-plugin/src/rules/space-infix-ops.ts

```diff
@@ -119,11 +128,38 @@ export default util.createRule<Options, MessageIds>({
       checkAndReportAssignmentSpace(node, leftNode, rightNode);
     }
 
+    /**
+     * Check if it is missing spaces between type annotations chaining
+     * @param typeAnnotation TypeAnnotations list
+     */
+    function checkForTypeAnnotationSpace(
+      typeAnnotation: TSESTree.TSIntersectionType | TSESTree.TSUnionType,
+    ): void {
+      const UNIONS = ['|', '&'];
```

Comment:
nitpick: please extract `UNIONS` into the root module scope, and make it a `Set`